Make "static" serving optional

Recently there was a change that now requires you have a `static` box in order to run `cfssl serve` otherwise it panics with:
```
panic: could not locate box "static"
```
Can you make that optional? I don't want to serve any static files I just want to serve is `/newcert` and I have no need for any static files. Older versions worked perfectly for this. It makes it hard to distribute a single static binary built with `go build` since there's now a new requirement that you have a static folder of assets.

--- a/cli/config.go
+++ b/cli/config.go
@@ -52,7 +52,6 @@ type Config struct {
 	Scanner           string
 	Responses         string
 	Path              string
-	UseLocal          bool
 	Usage             string
 }
 
@@ -92,7 +91,6 @@ func registerFlags(c *Config, f *flag.FlagSet) {
 	f.StringVar(&c.Responses, "responses", "", "file to load OCSP responses from")
 	f.StringVar(&c.Path, "path", "/", "Path on which the server will listen")
 	f.StringVar(&c.Password, "password", "0", "Password for accessing PKCS #12 data passed to bundler")
-	f.BoolVar(&c.UseLocal, "uselocal", false, "serve local static files as opposed to compiled ones")
 	f.StringVar(&c.Usage, "usage", "dev", "usage of private key")
 
 	if pkcs11.Enabled {

--- a/cli/serve/README.md
+++ b/cli/serve/README.md
@@ -9,5 +9,5 @@ esc -pkg serve -prefix cli/serve/static cli/serve/static > cli/serve/static.go
 
 # Build and run CFSSL
 go build ./cmd/cfssl/...
-./cfssl serve # -uselocal # Optionally use updating files on local filesystem
+./cfssl serve
 ```

--- a/cli/serve/serve.go
+++ b/cli/serve/serve.go
@@ -31,17 +31,18 @@ var serverUsageText = `cfssl serve -- set up a HTTP server handles CF SSL reques
 Usage of serve:
         cfssl serve [-address address] [-ca cert] [-ca-bundle bundle] \
                     [-ca-key key] [-int-bundle bundle] [-int-dir dir] [-port port] \
-                    [-metadata file] [-remote remote_host] [-config config] [-uselocal]
+                    [-metadata file] [-remote remote_host] [-config config]
 
 Flags:
 `
 
 // Flags used by 'cfssl serve'
-var serverFlags = []string{"address", "port", "ca", "ca-key", "ca-bundle", "int-bundle", "int-dir", "metadata", "remote", "config", "uselocal"}
+var serverFlags = []string{"address", "port", "ca", "ca-key", "ca-bundle", "int-bundle", "int-dir", "metadata", "remote", "config"}
 
 var (
-	conf cli.Config
-	s    signer.Signer
+	conf      cli.Config
+	s         signer.Signer
+	staticDir = "static"
 )
 
 var errBadSigner = errors.New("signer not initialized")
@@ -94,9 +95,23 @@ var v1Endpoints = map[string]func() (http.Handler, error){
 	"scaninfo": func() (http.Handler, error) {
 		return scan.NewInfoHandler(), nil
 	},
+}
 
+var staticEndpoints = map[string]func() (http.Handler, error){
 	"/": func() (http.Handler, error) {
-		return http.FileServer(rice.MustFindBox("static").HTTPBox()), nil
+		//the default rice order doesn't include working directory
+		order := []rice.LocateMethod{
+			rice.LocateEmbedded,
+			rice.LocateAppended,
+			rice.LocateFS,
+			rice.LocateWorkingDirectory,
+		}
+		c := &rice.Config{LocateOrder: order}
+		box, err := c.FindBox(staticDir)
+		if err != nil {
+			return nil, err
+		}
+		return http.FileServer(box.HTTPBox()), nil
 	},
 }
 
@@ -112,12 +127,16 @@ func v1APIPath(endpoint string) string {
 // registerHandlers instantiates various handlers and associate them to corresponding endpoints.
 func registerHandlers() {
 	for path, getHandler := range v1Endpoints {
-		if path != "/" {
-			path = "/api/v1/cfssl/" + path
-		}
-
+		path = "/api/v1/cfssl/" + path
 		log.Infof("Setting up '%s' endpoint", path)
-
+		if handler, err := getHandler(); err != nil {
+			log.Warningf("endpoint '%s' is disabled: %v", path, err)
+		} else {
+			http.Handle(path, handler)
+		}
+	}
+	for path, getHandler := range staticEndpoints {
+		log.Infof("Setting up '%s' endpoint", path)
 		if handler, err := getHandler(); err != nil {
 			log.Warningf("endpoint '%s' is disabled: %v", path, err)
 		} else {

--- a/cli/serve/serve_test.go
+++ b/cli/serve/serve_test.go
@@ -18,7 +18,6 @@ func TestServe(t *testing.T) {
 		expected[v1APIPath(endpoint)] = http.StatusOK
 	}
 
-	staticDir := "static"
 	err := rice.MustFindBox(staticDir).Walk("", func(path string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err

